packages/medusa/src/services/line-item.ts

```diff
@@ -182,12 +209,42 @@ class LineItemService extends BaseService {
     })
   }
 
+  /**
+   * Updates a line item
+   * @param {LineItem[]} lineItems - the id of the line item to update
+   * @param {ShippingMethod[]} shippingMethods - the properties to update on line item
+   * @return {Promise<LineItem[]>} the update line item
+   */
+  async updateHasShipping(
+    lineItems: LineItem[],
+    shippingMethods: ShippingMethod[]
+  ): Promise<LineItem[]> {
+    return this.atomicPhase_(async (manager) => {
+      const lineItemRepository: LineItemRepository =
+        manager.getCustomRepository(this.lineItemRepository_)
+
+      const shippingProfiles = shippingMethods.map(
+        (sm) => sm.shipping_option.profile_id
+      )
+
+      lineItems.forEach((li) => {
+        li.has_shipping =
+          shippingProfiles.find(
+            (sp) => sp === li.variant?.product?.profile_id
+          ) !== undefined
+      })
+
+      const result = await lineItemRepository.save(lineItems)
+      return result
+    })
+  }
```

Comment:
Could we make a custom repository method for this too?

In SQL I think something like this would get us in the right direction:

```
select li.id from "line_item" li
left join product_variant pv on pv.id = variant_id
left join product prod on prod.id = pv.product_id
where cart_id = 'cart_01FMHPESC9SJM83PAZFTKB1GWY' 
and profile_id not in (
	select profile_id from "shipping_method" sm
	left join shipping_option so on so.id = sm.shipping_option_id
	where cart_id = 'cart_01FMHPESC9SJM83PAZFTKB1GWY'
);
```

We can then update the line items with has shipping accordingly